pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -5146,78 +5146,75 @@ metadata:
   namespace: foo
 spec:
   results:
-  - description: pipeline result
-    name: result
-    value: $(tasks.a-task.results.aResult)
+    - description: pipeline result
+      name: result
+      value: $(tasks.a-task.results.a-Result)
+    - description: custom task pipeline result
+      name: custom-result
+      value: $(tasks.b-task.results.b-Result)
   tasks:
-  - name: a-task
-    taskRef:
-      name: a-task
   - name: b-task
-    params:
-    - name: bParam
-      value: $(tasks.a-task.results.aResult)
     taskRef:
+      apiVersion: example.dev/v0
+      kind: Example
       name: b-task
+  - name: a-task
+    taskRef:
+      name: a-task
 `)}
 	trs := []*v1beta1.TaskRun{mustParseTaskRunWithObjectMeta(t,
-		taskRunObjectMeta("test-pipeline-run-different-service-accs-a-task", "foo",
-			"test-pipeline-run-different-service-accs", "test-pipeline", "a-task", true),
+		taskRunObjectMeta("test-pipeline-run-results-task-run-a", "foo",
+			"test-pipeline-run-results", "test-pipeline", "a-task", true),
 		`
 spec:
-  resources: {}
-  serviceAccountName: test-sa
   taskRef:
     name: hello-world
-  timeout: 1h0m0s
 status:
   conditions:
   - status: "True"
     type: Succeeded
   taskResults:
-  - name: aResult
+  - name: a-Result
     value: aResultValue
 `)}
+
+	rs := []*v1alpha1.Run{mustParseRunWithObjectMeta(t,
+		taskRunObjectMeta("test-pipeline-run-results-task-run-b", "foo",
+			"test-pipeline-run-results", "test-pipeline", "b-task", true),
+		`
+spec:
+  ref:
+    apiVersion: example.dev/v0
+    kind: Example
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+  results:
+  - name: b-Result
+    value: bResultValue
+`)}
 	prs := []*v1beta1.PipelineRun{parse.MustParsePipelineRun(t, `
 metadata:
-  name: test-pipeline-run-different-service-accs
+  name: test-pipeline-run-results
   namespace: foo
 spec:
   pipelineRef:
     name: test-pipeline
-  serviceAccountName: test-sa-0
 status:
-  completionTime: "2022-01-01T00:00:00Z"
   conditions:
-  - message: All Tasks have completed executing
-    reason: Succeeded
-    status: "True"
+  - status: "Unknown"
     type: Succeeded
-  pipelineResults:
-  - name: result
-    value: aResultValue
-  startTime: "2021-12-31T00:00:00Z"
-  taskRuns:
-    test-pipeline-run-different-service-accs-a-task:
-      pipelineTaskName: a-task
-      status:
-        conditions:
-        - status: "True"
-          type: Succeeded
-        taskResults:
-        - name: aResult
-          value: aResultValue
+    reason: Succeeded
```

Comment:
we should change this `reason` to `"running"` because when the status is `"unknown"` it means that the pipelinerun is running....we could also just remove the `reason` field to keep it simple

Reply:
deleted to keep it simple, thank you for the feedback!